src/main/java/uk/ac/ic/wlgitbridge/bridge/db/postgres/PostgresDBStore.java

```diff
@@ -0,0 +1,298 @@
+    }
+  }
+
+  public void dangerousResetDatabaseDontDoIt() {
+    try {
+      Statement statement = connection.createStatement();
+      statement.execute("delete from url_index_store;");
+      statement.execute("delete from projects;");
+      statement.close();
+    } catch (Exception e) {
+      Log.error("Error while querying Postgres: {}", e.getMessage());
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public int getNumProjects() {
+    try {
+      Statement statement = connection.createStatement();
+      ResultSet rs = statement.executeQuery("SELECT count(*) from projects");
+      if (rs.next()) {
+        int result = rs.getInt(1);
+        rs.close();
+        statement.close();
+        return result;
+      } else {
+        throw new Exception("No results from query");
+      }
+    } catch (Exception e) {
+      Log.error("Error while querying Postgres: {}", e.getMessage());
+      throw new RuntimeException(e);
+    }
+  };
+
+  @Override
+  public List<String> getProjectNames() {
+    try {
+      Statement statement = connection.createStatement();
+      ResultSet rs = statement.executeQuery("SELECT name from projects");
+      List<String> result = new ArrayList<String>();
+      while (rs.next()) {
+        result.add(rs.getString(1));
+      }
+      rs.close();
+      statement.close();
+      return result;
+    } catch (Exception e) {
+      Log.error("Error while querying Postgres: {}", e.getMessage());
+      throw new RuntimeException(e);
+    }
+  };
+
+  @Override
+  public void setLatestVersionForProject(String project, int versionID) {
+    try {
+      PreparedStatement statement = connection.prepareStatement(
+        "INSERT INTO "
+          + "projects (name, version_id, last_accessed) "
+          + "VALUES (?, ?, now()) "
+          + "ON CONFLICT (name) DO UPDATE "
+          + "SET version_id = EXCLUDED.version_id, last_accessed = now();\n"
+      );
+      statement.setString(1, project);
+      statement.setInt(2, versionID);
+      statement.executeUpdate();
+      statement.close();
+      return;
+    } catch (Exception e) {
+      Log.error("Error while querying Postgres: {}", e.getMessage());
+      throw new RuntimeException(e);
+    }
+  };
+
+  @Override
+  public int getLatestVersionForProject(String project) {
+    try {
+      PreparedStatement statement = connection.prepareStatement(
+        "SELECT version_id from projects where name = ?;"
+      );
+      statement.setString(1, project);
+      ResultSet rs = statement.executeQuery();
+      if (rs.next()) {
+        int versionId = rs.getInt(1);
+        rs.close();
+        statement.close();
+        return versionId;
+      } else {
+        return 0;
+      }
+    } catch (Exception e) {
+      Log.error("Error while querying Postgres: {}", e.getMessage());
+      throw new RuntimeException(e);
+    }
+  };
+
+  @Override
+  public void addURLIndexForProject(String projectName, String url, String path) {
```

Comment:
The term `Index` is confusing here, but again it's in the base class and it's what the table is called, so I don't think it should be changed.